engine/api/workflow/dao.go

```diff
@@ -918,7 +918,22 @@ func Update(ctx context.Context, db gorp.SqlExecutor, store cache.Store, proj sd
 
 	// Delete workflow data
 	if uptOption.OldWorkflow != nil && uptOption.OldWorkflow.ID != 0 {
```

Comment:
not sure it is the best to fix the issue. I rather remove the pointers OldWorkflow and WorkflowData to a concrete types. To prevent having nil values - it should never happen

Reply:
done for oldWorkflow. For workflowData, too much impact, will be in next PR.